### PR TITLE
Merge branch 'combine-main' into master for better ease of development

### PR DESCRIPTION
**Pull request by peterfriese**
As discussed, let's merge combine-main into master to make ongoing development easier.

@lorenzofiamingo, @ziadtamim, and @mortenbekditlevsen FYI. Your working branched should update automatically once we close the `combine-main` branch.

Next steps:
1. Migrate tests to SwiftPM (@paulb777) 
    - Removing CocoaPods support
    - Make sure we're able to run the tests from SwiftPM
2. Need to verify we have test coverage across APIs
3. API review of existing public APIs + proposal to keep adding APIs under "community support" with @ryanwilson reviewing the new additions.

### CODE REVIEW

File: FirebaseCombineSwift/Sources/Auth/MultiFactor+Combine.swift
Comment (paulb777) [inline FirebaseCombineSwift/Sources/Auth/MultiFactor+Combine.swift:1]: I'd like to revisit the directory structure since I think we might prefer a product specific based directory structure like `FirebaseAuthCombine` to be consistent with the `FirebaseStorageSwift` structure already being used.

I'm fine to defer this refactor until after moving to master.

File: Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRHTTPSCallable.h
Comment (paulb777) [inline Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRHTTPSCallable.h:27]: @ryanwilson Does this need an API review?  Is there a workaround to do in the meantime? Is this part of a larger open question about exposing public initializers for testing?

File: Package.swift
Comment (paulb777) [inline Package.swift:42]: In our meeting, we stated that the new library products weren't needed - but I'm not sure about that.  Maybe we need to add a little tool to find/insert products into the Package.swift for development?

File: Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRHTTPSCallable.h
Comment (peterfriese) [reply Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRHTTPSCallable.h:27]: We might be able to replace this by using the Fakes infrastructure Maksym provided in `FirebaseTestingSupport`

File: Package.swift
Comment (peterfriese) [reply Package.swift:42]: Happy to remove them for now - LMK.

File: FirebaseCombineSwift/Sources/Auth/MultiFactor+Combine.swift
Comment (peterfriese) [reply FirebaseCombineSwift/Sources/Auth/MultiFactor+Combine.swift:1]: The reason for keeping everything under one umbrella folder was to make it easier to find the code (and any related documentation) that belongs together.

I can see benefits and drawbacks for both approaches. 

On a related note, would this refactoring be part of our endeavour to reduce the number of top-level folders (to bring the README.md above the fold), as suggested by @ncooke3 ?

Comment (paulb777) [reply FirebaseCombineSwift/Sources/Auth/MultiFactor+Combine.swift:1]: Understood  thanks. Let's defer this discussion until after the merge and the README.md and top-level folder structure plan is further along that can incorporate this open question.

File: Package.swift
Comment (paulb777) [reply Package.swift:42]: Perhaps, comment them out for now? @ryanwilson Any thoughts here?

**Comment by paulb777** [comment]
LGTM, pending two open questions for @ryanwilson

**Comment by paulb777** [comment]
BTW, I suggest doing the master merge from the command line to avoid squashing the branch history.

File: Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRHTTPSCallable.h
Comment (ryanwilson) [reply Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRHTTPSCallable.h:27]: Yes, this would need an API review but preferably it should use the same fakes infrastructure as `FirebaseTestingSupport` provides.

File: Package.swift
Comment (ryanwilson) [reply Package.swift:42]: Hmm, yeah commenting them out sounds good for now.

File: Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRHTTPSCallable.h
Comment (peterfriese) [reply Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRHTTPSCallable.h]: Rolled back opening up the API, and used a fake to implement the Cloud Functions tests.

File: FirebaseCombineSwift.podspec
Comment (paulb777) [inline FirebaseCombineSwift.podspec:22]: This is incorrect. Please update to use ruby variables like https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseMessaging.podspec#L23

File: FirebaseCombineSwift/Sources/Auth/MultiFactor+Combine.swift
Comment (paulb777) [reply FirebaseCombineSwift/Sources/Auth/MultiFactor+Combine.swift:1]: I think the compelling reason to change the organization will be the installation and build time of including a dependency on all Firebase products, especially Firestore. Let's sort out after the merge.

**Review by paulb777 — APPROVED** [review]
Thanks, this is close, and I'm fine if we address my remaining comments before or after merge.

Would you a short note to the top-level readme stating something like Combine support is under development and not available for production along with a pointer to the doc pages.

File: Package.swift
Comment (paulb777) [inline Package.swift:360]: I believe these targets should be deleted in favor of the commented ones above.

**Pull request merged**
